power_grid_model_c/power_grid_model/include/power_grid_model/index_mapping.hpp

```diff
@@ -125,12 +125,13 @@ inline auto build_dense_mapping_counting_sort(IdxVector const& idx_B_in_A, Idx c
 } // namespace detail
 
 inline DenseIndexMapping build_dense_mapping(IdxVector const& idx_B_in_A, Idx const n_B) {
-    constexpr auto relative_complexity_prefactor = 1.0;
-
     auto const n_A_ = static_cast<double>(idx_B_in_A.size());
     auto const n_B_ = static_cast<double>(n_B);
 
-    if (n_A_ + n_B_ < relative_complexity_prefactor * n_A_ * log(n_A_)) {
+    constexpr double coeff_1 = -0.62458071347;
+    constexpr double coeff_2 = -0.00038127964;
+
+    if (constexpr double coeff_3 = -996.744708763; n_A_ < coeff_1 * n_B_ + coeff_2 * n_A_ * log(n_A_) + coeff_3) {
```

Comment:
maybe more clear:

* rename `namespace detail` to `namespace index_mapping::detail`
* in `namespace index_mapping::detail`, create a struct
  ```cpp
  struct IndexMappingCriterion {
    double coeff_1{};
    double coeff_2{};
    double coeff_3{};

    constexpr bool operator()(std::integral auto n_A, std::integral auto n_B) const {
        auto const n_A_ = static_cast<double>(n_A);
        auto const n_B_ = static_cast<double>(n_B);
        return n_A < coeff_1 * n_B_ + coeff_2 * n_A_ * log(n_A_) + coeff_3;
    }
  };

  constexpr IndexMappingCriterion index_mapping_criterion_gcc{.coeff_1 = -0.62458071347, .coeff_2 = -0.00038127964, .coeff_3 = -996.744708763};
  ```

```suggestion
    if (index_mapping_criterion_gcc(idx_B_in_A.size(), n_B)) {
```